pkg/provider/provider.go

```diff
@@ -101,23 +112,57 @@ func DSN(s *schema.ResourceData) (string, error) {
 		region = ""
 	}
 
-	dsn, err := gosnowflake.DSN(&gosnowflake.Config{
-		Account:  account,
-		User:     username,
-		Region:   region,
-		Password: password,
-		Role:     role,
-	})
-
-	if browserAuth {
-		dsn, err = gosnowflake.DSN(&gosnowflake.Config{
-			Account:       account,
-			User:          username,
-			Region:        region,
-			Role:          role,
-			Authenticator: gosnowflake.AuthTypeExternalBrowser,
-		})
+	config := gosnowflake.Config{
+		Account: account,
+		User:    user,
+		Region:  region,
+		Role:    role,
 	}
 
-	return dsn, err
+	if len(privateKeyPath) != 0 {
```

Comment:
you can also do 
```suggestion
	if privateKeyPath != "" {
```
since the 0 value for strings is `""`